tests/deepspeed/test_deepspeed.py

```diff
@@ -72,6 +75,27 @@
 GPT2_TINY = "sshleifer/tiny-gpt2"
 
 
+class TinyGPT2WithUninitializedWeights(GPT2PreTrainedModel):
```

Comment:
should be defined direclty in the test as the test requires torch

Reply:
(it's a nit but that way you only import everything once, if torch is available, then the test requires torch_accelerator so will be ran only if torch is available

Reply:
+1  on these test and import comments - we need to make sure things are safely imported and only when needed

Reply:
Thanks for the comment. I moved the imports and class definition to the test case body only.